nocgo.go

```diff
@@ -0,0 +1,12 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2022 The Ebitengine Authors
+
+//go:build !cgo && !ios
+// +build !cgo,!ios
+
+package purego
+
+// if CGO_ENABLED=0 import fakecgo to setup the Cgo runtime correctly.
+// This is required since some frameworks need TLS setup the C way which Go doesn't do.
+// We currently don't support ios in fakecgo mode so force Cgo or fail
+import _ "github.com/ebitengine/purego/internal/fakecgo"
```

Comment:
I'm not sure why just importing this package works (this is magical!) Could you elaborate this?

Reply:
My understanding is that fakecgo does what `runtime/cgo` does instead of it. Is this correct?

Reply:
So, does fakecgo creates these symbols instead of runtime/cgo?

https://cs.opensource.google/go/go/+/master:src/runtime/cgo.go;l=13-19;drc=2580d0e08d5e9f979b943758d3c49877fb2324cb

Reply:
Yeah that's the basic idea. All you need to do is `go:linkname` each of those symbols with a Go version (well, asm since you need to map from GCC ABI to Go ABI). We don't replace all of them just the one's listed below:
```go
iscgo             bool
_cgo_init         unsafe.Pointer
_cgo_thread_start unsafe.Pointer
_cgo_notify_runtime_init_done unsafe.Pointer
_cgo_setenv unsafe.Pointer  // these two are in runtime/env_posix.go
_cgo_unsetenv unsafe.Pointer
```
[notti/nocgo](https://github.com/notti/nocgo#fakecgo) contains a pretty good explanation of how his fakecgo works which this PR is inspired by.

Reply:
It's just enough to pass this `iscgo` test in [runtime/proc.go](https://github.com/golang/go/blob/c82bbc0e8edbbebe47e92729e8f3f1b60d380b5b/src/runtime/proc.go#L212) and setup thread-local storage properly.